metacat-common-server/src/main/java/com/netflix/metacat/common/server/events/MetacatApplicationEventMulticaster.java

```diff
@@ -54,27 +54,23 @@ public MetacatApplicationEventMulticaster(
     }
 
     /**
-     * Post event asynchronously.
+     * Post event. Events will be handles synchronously or asynchronously based on the listener annotation.
      *
      * @param event event
      */
-    public void postAsync(final ApplicationEvent event) {
-        this.asyncEventMulticaster.multicastEvent(event);
-    }
-
-    /**
-     * Post event synchronously.
-     *
-     * @param event event
-     */
-    public void postSync(final ApplicationEvent event) {
+    public void post(final ApplicationEvent event) {
         super.multicastEvent(event);
+        this.asyncEventMulticaster.multicastEvent(event);
     }
```

Comment:
Why multicasts twice here?

Reply:
Earlier the publishers were explicitly calling async or sync methods. Now I have changed to make the listeners register as sync or async.

Reply:
If I understand right. The async multicast will happen after executing the listeners in sync multicast, say if one if the action failed and the async multicast won't happen. I guess this is what you want, is just the code looks a bit weird.